### PR TITLE
Convert registry to k8s.gcr.io

### PR DESCRIPTION
**Pull request by thockin**
This PR was auto-generated.  Please apply human expertise to review for correctness.

Followup to https://github.com/kubernetes/kubernetes/pull/54174 and https://github.com/kubernetes/kubernetes/pull/57824

xref https://github.com/kubernetes/release/issues/281

### CODE REVIEW

**Comment by bowei** [comment]
lgtm

**Review by MrHohn — APPROVED** [review]
/lgtm

**Review by porridge — CHANGES_REQUESTED** [review]
Sorry, but this does not work.
I can push to the old location but not the new one:

```
porridge@kielonek:~/projects/go/src/k8s.io/contrib/perfdash$ gcloud docker -- tag gcr.io/google_containers/perfdash:0.8 k8s.gcr.io/perfdash:0.8
porridge@kielonek:~/projects/go/src/k8s.io/contrib/perfdash$ gcloud docker -- push  k8s.gcr.io/perfdash:0.8
The push refers to a repository [k8s.gcr.io/perfdash]
45f4f6715eed: Pushing [==================================================>] 137.2 kB
eacb8c6cad16: Pushing [==========================>                        ] 2.819 MB/5.321 MB
df6868d54704: Pushing [=====================>                             ] 2.805 MB/6.572 MB
8880784f79cf: Pushing [====>                                              ] 3.518 MB/36.78 MB
46ecc6c4543a: Pushing [>                                                  ] 2.126 MB/163.5 MB
5f70bf18a086: Waiting 
unexpected EOF
# (The above retries several times before failing)
porridge@kielonek:~/projects/go/src/k8s.io/contrib/perfdash$ gcloud docker -- tag gcr.io/google_containers/perfdash:0.8 gcr.io/google_containers/perfdash:0.8-test
porridge@kielonek:~/projects/go/src/k8s.io/contrib/perfdash$ gcloud docker -- push  gcr.io/google_containers/perfdash:0.8-test
The push refers to a repository [gcr.io/google_containers/perfdash]
45f4f6715eed: Layer already exists 
eacb8c6cad16: Layer already exists 
df6868d54704: Layer already exists 
8880784f79cf: Layer already exists 
46ecc6c4543a: Layer already exists 
5f70bf18a086: Layer already exists 
0.8-test: digest: sha256:b1ed038f1e032ea33fb48672d779f96a4c225140381ec8ba7ee29648dae2981f size: 1783
```

**Comment by thockin** [comment]
@dekkagaijin  I, too, am experiencing issues.  This all worked at some point.

```
thockin@freakshow2 kubernetes gcr-vanity /$ docker images | egrep '(pause|thtest)'

thockin@freakshow2 kubernetes gcr-vanity /$ docker pull gcr.io/google_containers/pause
Using default tag: latest
latest: Pulling from google_containers/pause
a3ed95caeb02: Pull complete 
4964c72cd024: Pull complete 
Digest: sha256:a78c2d6208eff9b672de43f880093100050983047b7b0afe0217d3656e1b0d5f
Status: Downloaded newer image for gcr.io/google_containers/pause:latest

thockin@freakshow2 kubernetes gcr-vanity /$ docker images | egrep '(pause|thtest)'
gcr.io/google_containers/pause                   latest                        350b164e7ae1        3 years ago         239.8 kB

thockin@freakshow2 kubernetes gcr-vanity /$ docker tag gcr.io/google_containers/pause k8s.gcr.io/thtest:thtest 

thockin@freakshow2 kubernetes gcr-vanity /$ docker images | egrep '(pause|thtest)'
gcr.io/google_containers/pause                   latest                        350b164e7ae1        3 years ago         239.8 kB
k8s.gcr.io/thtest                                thtest                        350b164e7ae1        3 years ago         239.8 kB

thockin@freakshow2 kubernetes gcr-vanity /$ gcloud docker -- push k8s.gcr.io/thtest:thtest
The push refers to a repository [k8s.gcr.io/thtest]
5f70bf18a086: Layer already exists 
e16a89738269: Pushing [==================================================>] 241.7 kB
unexpected EOF
```

**Comment by dekkagaijin** [comment]
2 possible solutions: updated gcloud to >= 183.0.0 (where k8s.gcr.io was added as a default domain) or use `gcloud docker -s k8s.gcr.io -- push ...`

**Comment by thockin** [comment]
Confirmed, update worked for me

**Review by porridge — APPROVED** [review]
The `-s` flag worked as well.
/lgtm

File: for-demos/proxy-to-service/Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@
 .PHONY: all container push
 
 TAG = v2
-PREFIX = gcr.io/google_containers
+PREFIX = k8s.gcr.io
```
Comment (dekkagaijin) [inline for-demos/proxy-to-service/Makefile]: `REGISTRY = staging-k8s.gcr.io`

...

```
container:
	docker build --pull -t $(REGISTRY)/$(NAME):$(TAG) .

push:
	gcloud docker -s REGISTRY-- push $(REGISTRY)/$(NAME):$(TAG)
```

File: images/haproxy/Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ all: push
 
 # 0.0.0 shouldn't clobber any released builds
 TAG = 0.5
-PREFIX = gcr.io/google_containers/haproxy
+PREFIX = k8s.gcr.io/haproxy
```
Comment (dekkagaijin) [inline images/haproxy/Makefile]:
```
REGISTRY = staging-k8s.gcr.io
PREFIX = $(REGISTRY)/haproxy
```

...

```
push: container
	gcloud docker -s $(REGISTRY) -- push $(PREFIX):$(TAG)
```

File: ingress/echoheaders-redirect/Makefile [hunk 3]
```diff
@@ -2,7 +2,7 @@ all: push
 
 # TAG 0.0 shouldn't clobber any release builds
 TAG = 1.3
-PREFIX = gcr.io/google_containers/echoserver-redirect
+PREFIX = k8s.gcr.io/echoserver-redirect
```
Comment (dekkagaijin) [inline ingress/echoheaders-redirect/Makefile]:
```
REGISTRY = staging-k8s.gcr.io
PREFIX = $(REGISTRY)/echoserver-redirect

container:
	docker build --pull -t $(PREFIX):$(TAG) .

push: container
	gcloud docker -s $(REGISTRY) -- push $(PREFIX):$(TAG)
```

File: keepalived-vip/Makefile [hunk 4]
```diff
@@ -2,7 +2,7 @@ all: push
 
 # 0.0 shouldn't clobber any release builds, current "latest" is 0.11
 TAG = 0.11
-PREFIX = gcr.io/google_containers/kube-keepalived-vip
+PREFIX = k8s.gcr.io/kube-keepalived-vip
```
Comment (dekkagaijin) [inline keepalived-vip/Makefile]:
```
REGISTRY = staging-k8s.gcr.io
PREFIX = $(REGISTRY)/kube-keepalived-vip
```
...
```
push: container
	gcloud docker -s $(REGISTRY) -- push $(PREFIX):$(TAG)
```

File: node-perf-dash/Makefile [hunk 5]
```diff
@@ -3,7 +3,7 @@ all: push
 # See pod.yaml for the version currently running-- bump this ahead before rebuilding!
 TAG = v0.3
 
-REPO = gcr.io/google_containers
+REPO = k8s.gcr.io
```
Comment (dekkagaijin) [inline node-perf-dash/Makefile]:
```
REGISTRY = staging-k8s.gcr.io
REPO = $(REGISTRY)

...

push: container
	gcloud docker -s $(REGISTRY) -- push $(REPO)/node-perf-dash:$(TAG)

File: peer-finder/Makefile [hunk 6]
```diff
@@ -15,7 +15,7 @@
 all: push
 
 TAG = 0.1
-PREFIX = gcr.io/google_containers/peer-finder
+PREFIX = k8s.gcr.io/peer-finder
```
Comment (dekkagaijin) [inline peer-finder/Makefile]:
```
REGISTRY = staging-k8s.gcr.io
PREFIX = $(REGISTRY)/peer-finder

...

push: container
	gcloud docker -s $(REGISTRY) -- push $(PREFIX):$(TAG)

File: perfdash/Makefile [hunk 7]
```diff
@@ -3,16 +3,16 @@ all: push
 # See pod.yaml for the version currently running-- bump this ahead before rebuilding!
 TAG = 0.8
 
-PROJ = google_containers
+REPO = k8s.gcr.io
```
Comment (dekkagaijin) [inline perfdash/Makefile]: REGISTRY = staging-k8s.gcr.io
REPO = $(REGISTRY)

...

push: container
	gcloud docker -s $(REGISTRY) -- push $(REPO)/perfdash:$(TAG)

File: scale-demo/vegeta/Makefile [hunk 8]
```diff
@@ -7,10 +7,10 @@ loader: loader.go
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 godep go build -a -installsuffix cgo -ldflags '-w' -o loader
 
 container: loader
-	docker build --pull -t gcr.io/google_containers/loader:$(TAG) .
+	docker build --pull -t k8s.gcr.io/loader:$(TAG) .
 
 push: container
-	gcloud docker -- push gcr.io/google_containers/loader:$(TAG)
+	gcloud docker -- push k8s.gcr.io/loader:$(TAG)
```
Comment (dekkagaijin) [inline scale-demo/vegeta/Makefile]: `gcloud docker -s staging-k8s.gcr.io -- push staging-k8s.gcr.io/loader:$(TAG)`

File: service-loadbalancer/Makefile [hunk 9]
```diff
@@ -2,7 +2,7 @@ all: push
 
 # 0.0 shouldn't clobber any released builds
 TAG ?= 0.4
-PREFIX ?= gcr.io/google_containers/servicelb
+PREFIX ?= k8s.gcr.io/servicelb
```
Comment (dekkagaijin) [inline service-loadbalancer/Makefile]:
```
REGISTRY ?= staging-k8s.gcr.io
PREFIX ?= $(REGISTRY)/servicelb

...

push: container
	$(GCLOUD) docker -s $(REGISTRY) -- push $(PREFIX):$(TAG)
```

**Review by dekkagaijin — CHANGES_REQUESTED** [review]

**Pull request merged**